src/main/java/com/example/demo/controller/ProductController.java

```diff
@@ -0,0 +1,60 @@
+package com.example.demo.controller;
+
+import com.example.demo.model.Product;
+import com.example.demo.repository.ProductRepository;
+import com.example.demo.repository.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.Set;
+
+@Controller
+public class ProductController {
+
+//    private ProductRepository productRepository;
+//
+//    @Autowired
+//    public ProductController(ProductRepository productRepository) {
+//        this.productRepository = productRepository;
+//    }
+
+    private ProductService productService;
+
+    @Autowired
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping("/lista")
+    public String getProductsList(Model model) {
+        Set<Product> products;
+        products = productService.findAll();
```

Comment:
Nie musiałeś tego rozdzielać na definicję zmiennej i przypisanie. Można to zrobić w jednej linii